dht.go

```diff
@@ -223,7 +223,38 @@ func makeDHT(ctx context.Context, h host.Host, cfg opts.Options) *IpfsDHT {
 
 	dht.providers = providers.NewProviderManager(dht.ctx, h.ID(), cfg.Datastore)
 
-	return dht
+	return dht, nil
+}
+
+func makeRoutingTable(h host.Host, cfg *opts.Options) (*kb.RoutingTable, error) {
```

Comment:
Any reason we're using `*opts.Options` instead of `opts.Options`? Seems like we may as well be consistent with `makeDHT`, so I'd make them both use either pointer or value semantics. Since we're only doing gets on a small data structure I figured use value semantics, but I could be swayed if you think this is better.

Reply:
I don't care either ways. Let's go with opts.Options.

Reply:
Done.